posthog/hogql/resolver.py

```diff
@@ -275,6 +275,11 @@ def visit_join_expr(self, node: ast.JoinExpr):
                 return response
 
         if isinstance(node.table, ast.Field):
+            node = cast(ast.JoinExpr, clone_expr(node))
+            if node.constraint and node.constraint.constraint_type == "USING":
+                # visit USING constraint before adding the table to avoid ambiguous names
+                node.constraint = self.visit_join_constraint(node.constraint)
```

Comment:
`visit_join_constraint` instead of `visit` to have correct types - visit returns Expr while visit_join_contraint returns JoinContraint